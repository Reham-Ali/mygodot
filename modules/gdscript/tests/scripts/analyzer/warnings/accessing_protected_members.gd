class A:
	@warning_ignore("unused_protected_class_variable")
	var _a
	static var _b

	func _c():
		pass
	static func _d():
		pass

class B:
	var a = A.new()
	var b = a._a
	var c = A._b

	func foo():
		@warning_ignore("standalone_expression")
		a._a
		@warning_ignore("standalone_expression")
		A._b

		a._c()
		A._d()

	var d = a._c()
	var e = A._d()

class C extends A:
	var a = A.new()
	var b = a._a
	var c = A._b

	func foo():
		@warning_ignore("standalone_expression")
		a._a
		@warning_ignore("standalone_expression")
		A._b

		a._c()
		A._d()

	var d = a._c()
	var e = A._d()

func test():
	pass
