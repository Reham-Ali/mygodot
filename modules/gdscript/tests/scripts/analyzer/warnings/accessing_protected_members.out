GDTEST_OK
>> WARNING
>> Line: 13
>> ACCESSING_PROTECTED_MEMBER
>> Trying accessing a protected member "_a" in an class that is not authorized to access it.
>> WARNING
>> Line: 14
>> ACCESSING_PROTECTED_MEMBER
>> Trying accessing a protected member "_b" in an class that is not authorized to access it.
>> WARNING
>> Line: 18
>> ACCESSING_PROTECTED_MEMBER
>> Trying accessing a protected member "_a" in an class that is not authorized to access it.
>> WARNING
>> Line: 20
>> ACCESSING_PROTECTED_MEMBER
>> Trying accessing a protected member "_b" in an class that is not authorized to access it.
>> WARNING
>> Line: 22
>> CALLING_PROTECTED_METHOD
>> Trying calling a protected method "_c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 23
>> CALLING_PROTECTED_METHOD
>> Trying calling a protected method "_d()" in an class that is not authorized to call it.
>> WARNING
>> Line: 25
>> CALLING_PROTECTED_METHOD
>> Trying calling a protected method "_c()" in an class that is not authorized to call it.
>> WARNING
>> Line: 26
>> CALLING_PROTECTED_METHOD
>> Trying calling a protected method "_d()" in an class that is not authorized to call it.
